shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceBeanPostProcessor.java

```diff
@@ -168,6 +179,39 @@ private MetaDataRegisterDTO buildMetaDataDTO(final Class<?> clazz, final MotanSe
                 .build();
     }
 
+    private MetaDataRegisterDTO buildMetaDataDTO(final Class<?> clazz, final MotanService service,
+                                                 final ShenyuMotanClient shenyuMotanClient, final String path) {
+        String appName = this.appName;
+        String desc = shenyuMotanClient.desc();
+        String host = IpUtils.isCompleteHost(this.host) ? this.host : IpUtils.getHost(this.host);
+        int port = StringUtils.isBlank(this.port) ? -1 : Integer.parseInt(this.port);
+        String configRuleName = shenyuMotanClient.ruleName();
+        String ruleName = ("".equals(configRuleName)) ? path : configRuleName;
+        String serviceName;
+        if (void.class.equals(service.interfaceClass())) {
+            if (clazz.getInterfaces().length > 0) {
+                serviceName = clazz.getInterfaces()[0].getName();
+            } else {
+                throw new ShenyuClientIllegalArgumentException("Failed to export remote service class " + clazz.getName()
+                        + ", cause: The @Service undefined interfaceClass or interfaceName, and the service class unimplemented any interfaces.");
+            }
+        } else {
+            serviceName = service.interfaceClass().getName();
+        }
+        return MetaDataRegisterDTO.builder()
```

Comment:
Lost `rpcExt`, which will cause an exception in the call of the `MotanPlugin`. The `metadata` of `motan` and `tars` client seems wrong. I'm discussing it with other developers. Please wait until I have a conclusion before continuing the development.